packages/curve-ui-kit/src/features/user-profile/home/Header.tsx

```diff
@@ -0,0 +1,47 @@
+import { styled } from '@mui/material/styles'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import IconButton from '@mui/material/IconButton'
+
+import SettingsIcon from '@mui/icons-material/Settings'
+import CloseIcon from '@mui/icons-material/Close'
+
+import { LlamaImg } from 'ui/src/images'
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+import { ConnectWalletIndicatorProps } from 'curve-common/src/features/connect-wallet'
+
+type Props = {
+  walletAddress: NonNullable<ConnectWalletIndicatorProps['walletAddress']>
+  onClose: () => void
+}
+
+const LlamaImageSrc = (LlamaImg as unknown as { src: string }).src
+const Llama = styled('img')({
+  height: '40px',
+})
+
+/**
+ * Gap deviates from Figma as personally 'xs' is too narrow.
+ *
+ * I know there's <AddressLabel> but not sure how to use it with a different typography,
```

Comment:
hmm weird one, indeed :thinking:
at least we should extract the substringing to a shared function imo
that AddressLabel is too simple and specific anyway it could be just inlined